packages/components/src/engine/metadata.ts

```diff
@@ -1,52 +1,106 @@
 import type { IsomerPageSchemaType, IsomerSitemap } from "~/types"
+import { ISOMER_PAGE_LAYOUTS } from "~/types"
 import { getSitemapAsArray } from "~/utils"
 
+const getMetaDescription = (props: IsomerPageSchemaType) => {
+  if (props.meta?.description) {
+    return props.meta.description
+  }
+
+  switch (props.layout) {
+    case ISOMER_PAGE_LAYOUTS.Article:
+      return props.page.articlePageHeader.summary
+    case ISOMER_PAGE_LAYOUTS.Content:
+    case ISOMER_PAGE_LAYOUTS.Database:
+    case ISOMER_PAGE_LAYOUTS.Index:
+      return props.page.contentPageHeader.summary
+    case ISOMER_PAGE_LAYOUTS.Collection:
+      return props.page.subtitle
+    case ISOMER_PAGE_LAYOUTS.Homepage:
+      return props.content.find((item) => item.type === "hero")?.subtitle
+    case ISOMER_PAGE_LAYOUTS.File:
+    case ISOMER_PAGE_LAYOUTS.Link:
+    case ISOMER_PAGE_LAYOUTS.Search:
+    case ISOMER_PAGE_LAYOUTS.NotFound:
+      // NOTE: These pages do not appear in search results, so we don't need to
+      // provide a meta description
+      return undefined
+    default:
+      const _: never = props
+      return undefined
+  }
+}
+
+const getMetaImage = (props: IsomerPageSchemaType) => {
+  switch (props.layout) {
+    case ISOMER_PAGE_LAYOUTS.Article:
+      return props.meta?.image || props.page.image?.src
+    case ISOMER_PAGE_LAYOUTS.Content:
+    case ISOMER_PAGE_LAYOUTS.Database:
+    case ISOMER_PAGE_LAYOUTS.Index:
+    case ISOMER_PAGE_LAYOUTS.Collection:
+      return props.meta?.image
+    case ISOMER_PAGE_LAYOUTS.Homepage:
+      return (
+        props.meta?.image ||
+        props.content.find((item) => item.type === "hero")?.backgroundUrl
+      )
+    case ISOMER_PAGE_LAYOUTS.File:
+    case ISOMER_PAGE_LAYOUTS.Link:
+    case ISOMER_PAGE_LAYOUTS.Search:
+    case ISOMER_PAGE_LAYOUTS.NotFound:
+      // NOTE: These pages do not appear in search results, so we don't need to
+      // provide a meta description
+      return undefined
+    default:
+      const _: never = props
+      return undefined
+  }
+}
+
```

Comment:
```suggestion
const getMetaImage = (props: IsomerPageSchemaType) => {
  if (props.meta?.image) {
    return props.meta.image
  }

  switch (props.layout) {
    case ISOMER_PAGE_LAYOUTS.Article:
      return props.page.image?.src
    case ISOMER_PAGE_LAYOUTS.Content:
    case ISOMER_PAGE_LAYOUTS.Database:
    case ISOMER_PAGE_LAYOUTS.Index:
    case ISOMER_PAGE_LAYOUTS.Collection:
      return undefined
    case ISOMER_PAGE_LAYOUTS.Homepage:
      return props.content.find((item) => item.type === "hero")?.backgroundUrl
    case ISOMER_PAGE_LAYOUTS.File:
    case ISOMER_PAGE_LAYOUTS.Link:
    case ISOMER_PAGE_LAYOUTS.Search:
    case ISOMER_PAGE_LAYOUTS.NotFound:
      // NOTE: These pages do not appear in search results, so we don't need to
      // provide a meta description
      return undefined
    default:
      const _: never = props
      return undefined
  }
}

```

Reply:
Hmm no, I see where you're coming from but we can't do this. The file and link layouts do not have support for meta images, so the attribute will not be present.

Actually in fact I think even for meta descriptions they should be removed.

Reply:
if the attribute isn't present, won't `props.meta?.image` be false and skip through that check?

Reply:
It's typescript that will not allow it, because the attribute doesn't exist on the link and file layouts so there will be a type error.